homeassistant/components/recorder/core.py

```diff
@@ -1236,9 +1236,31 @@ def _handle_sqlite_corruption(self, setup_run: bool) -> None:
         move_away_broken_database(dburl_to_path(self.db_url))
         self.recorder_runs_manager.reset()
         self._setup_recorder()
+        self._notify_db_corruption()
+
         if setup_run:
             self._setup_run()
 
+    def _notify_db_corruption(self) -> None:
+        """Notify users of SQLite database corruption.
+
+        Create a persistent notification and fire a 'recorder_database_corrupt' event to inform
+        users that their recorder database has been rebuilt due to database corruption.
+        """
+        persistent_notification.create(
```

Comment:
We consider persistent notifications to be user space nowadays, we should not be using that to send users system messages.

Reply:
Hey @frenck ,

Is there another mechanism we can use to display these sort of errors in the UI?

I can see value in there being some sort of proactive alert/notification as the database corruption is effectively handled silently, so users would not know to check the logs for the error.

Best,
Ryan

Reply:
> Is there another mechanism we can use to display these sort of errors in the UI?

The logs are currently the sole location for these things in that case.